src/containers/NFT/NFTHeader/index.js

```diff
@@ -0,0 +1,246 @@
+import Tooltip from '../../shared/components/Tooltip';
+import Copy from '../../shared/components/Copy';
+
+class NFTHeader extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  componentDidMount() {
+    const { actions, tokenId } = this.props;
+    const rippledSocket = this.context;
+    actions.loadNFTState(tokenId, rippledSocket.clioSocket);
+  }
+
+  componentDidUpdate(prevProps) {
+    const nexttokenId = prevProps.tokenId;
+    const { tokenId, actions } = this.props;
+    const rippledSocket = this.context;
+
+    if (nexttokenId !== tokenId) {
+      actions.loadNFTState(nexttokenId, rippledSocket.clioSocket);
+    }
+  }
+
+  showTooltip = (event, data) => {
+    this.setState({
+      tooltip: { ...data, mode: 'nftId', x: event.pageX, y: event.pageY },
+    });
+  };
+
+  hideTooltip = () => {
+    this.setState({ tooltip: null });
+  };
+
+  renderDetails() {
+    const { t, data } = this.props;
+    const { minted, domain, emailHash, NFTTaxon, uri, transferFee } = data;
+    const abbrvEmail = emailHash && emailHash.replace(/(.{20})..+/, '$1...');
+    const abbrvURI = uri && uri.replace(/(.{20})..+/, '$1...');
+    return (
+      <table>
+        <tbody>
+          {minted && (
+            <tr className="row">
+              <td className="col1">Minted</td>
+              <td className="col2">{minted}</td>
+            </tr>
+          )}
+          {domain && (
+            <tr className="row">
+              <td className="col1">{t('domain')}</td>
+              <td className="col2">
+                <a href={`https://${domain}`} target="_blank" rel="noopener noreferrer">
+                  {domain}
+                </a>
+              </td>
+            </tr>
+          )}
+          {emailHash && (
+            <tr className="row">
+              <td className="col1">{t('email_hash')}</td>
+              <td className="col2">
+                {abbrvEmail}
+                <Copy className="copy" text={emailHash} />
+              </td>
+            </tr>
+          )}
+          {NFTTaxon !== undefined && (
+            <tr className="row">
+              <td className="col1">Taxon ID</td>
+              <td className="col2">{NFTTaxon}</td>
+            </tr>
+          )}
+          {uri && (
+            <tr className="row">
+              <td className="col1">URI</td>
+              <td className="col2">
+                {abbrvURI}
+                <Copy className="copy" text={uri} />
+              </td>
+            </tr>
+          )}
+          {transferFee !== undefined && (
+            <tr className="row">
+              <td className="col1">Transfer Fee</td>
+              <td className="col2">{transferFee}</td>
+            </tr>
+          )}
+        </tbody>
+      </table>
+    );
+  }
+
+  renderSettings() {
+    const { data } = this.props;
+    const { flags } = data;
+
+    const burnable = flags && flags.includes('lsfBurnable') ? 'enabled' : 'disabled';
```

Comment:
flags should always be present from the API

Reply:
its being parsed by returning an array of enabled flags. The ones that are disabled wont be inside the array, thus why the condition check

Reply:
i get that, i'm saying why can't it be `const burnable = flags.includes('lsfBurnable') ? 'enabled': 'disabled'`

Reply:
the flags state is undefined until the ws request received the data.